src/mscorlib/corefx/System/Buffers/TlsOverPerCoreLockedStacksArrayPool.Unix.cs

```diff
@@ -0,0 +1,28 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Win32;
+using System.Runtime.CompilerServices;
+
+namespace System.Buffers
+{
+    internal sealed partial class TlsOverPerCoreLockedStacksArrayPool<T>
+    {
+        /// <summary>Get an identifier for the current thread to use to index into the stacks.</summary>
+        private static int ExecutionId
+        {
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            get
+            {
+                // On Unix, GetCurrentProcessorNumber is implemented in terms of sched_getcpu, which
```

Comment:
Similar problem may exist on Windows - some of combinations of old OS versions; old CPU versions; or some types of virtualized environments are likely going to have slow implementation of GetCurrentProcessorNumber.